src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4876,7 +4876,7 @@ void MergeTreeData::checkAlterPartitionIsPossible(
                 const auto * partition_ast = command.partition->as<ASTPartition>();
                 if (partition_ast && partition_ast->all)
                 {
-                    if (command.type != PartitionCommand::DROP_PARTITION && command.type != PartitionCommand::ATTACH_PARTITION)
+                    if (command.type != PartitionCommand::DROP_PARTITION && command.type != PartitionCommand::ATTACH_PARTITION && (command.type == PartitionCommand::REPLACE_PARTITION && command.replace))
```

Comment:
Why?

Reply:
There is a nuance: `ATTACH FROM` is similar to `REPLACE`, so they are merged into one function. When you query `ATTACH FROM`, ClickHouse executes `REPLACE` with replace=false, which means we should only copy partitions without deletion. Thus, if replace=true, it is `REPLACE` statement, which do not support `ALL` at the moment.